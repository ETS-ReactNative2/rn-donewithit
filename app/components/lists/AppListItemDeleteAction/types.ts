export type AppListItemDeleteActionProps = {
  onDelete?: () => void
}
