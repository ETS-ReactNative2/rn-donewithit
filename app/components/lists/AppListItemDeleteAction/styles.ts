import { StyleSheet } from 'react-native'
import COLORS from '../../../config/colors'

const styles = StyleSheet.create({
  container: {
    width: 80,
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: COLORS.DANGER
  }
})

export default styles
