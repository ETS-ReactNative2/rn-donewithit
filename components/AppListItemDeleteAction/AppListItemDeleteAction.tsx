import React, { useCallback } from 'react'
import { StyleSheet } from 'react-native'
import { View } from 'react-native'
import COLORS from '../../config/colors'
import CONSTS from './consts'
import { MaterialCommunityIcons } from '@expo/vector-icons'
import { TouchableWithoutFeedback } from 'react-native-gesture-handler'
import { AppListItemDeleteActionProps } from './types'

function AppListItemDeleteAction({ onDelete }: AppListItemDeleteActionProps) {
  const handlePress = useCallback(() => {
    typeof onDelete === 'function' && onDelete()
  }, [])

  return (
    <TouchableWithoutFeedback onPress={handlePress}>
      <View style={styles.container}>
        <MaterialCommunityIcons
          name="trash-can"
          size={CONSTS.ICON_SIZE}
          color={COLORS.WHITE}
        />
      </View>
    </TouchableWithoutFeedback>
  )
}

const styles = StyleSheet.create({
  container: {
    width: 80,
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: COLORS.DANGER
  }
})

export default AppListItemDeleteAction
