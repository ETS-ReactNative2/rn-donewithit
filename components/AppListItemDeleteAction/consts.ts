const CONSTS = {
  ICON_SIZE: 38
}

export default CONSTS
