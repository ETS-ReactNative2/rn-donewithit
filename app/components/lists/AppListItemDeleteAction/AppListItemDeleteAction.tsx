import React, { useCallback } from 'react'
import { View } from 'react-native'
import COLORS from '../../../config/colors'
import CONSTS from './consts'
import { MaterialCommunityIcons } from '@expo/vector-icons'
import { TouchableWithoutFeedback } from 'react-native-gesture-handler'
import { AppListItemDeleteActionProps } from './types'
import styles from './styles'

function AppListItemDeleteAction({ onDelete }: AppListItemDeleteActionProps) {
  const handlePress = useCallback(() => {
    typeof onDelete === 'function' && onDelete()
  }, [])

  return (
    <TouchableWithoutFeedback onPress={handlePress}>
      <View style={styles.container}>
        <MaterialCommunityIcons
          name="trash-can"
          size={CONSTS.ICON_SIZE}
          color={COLORS.WHITE}
        />
      </View>
    </TouchableWithoutFeedback>
  )
}

export default AppListItemDeleteAction
