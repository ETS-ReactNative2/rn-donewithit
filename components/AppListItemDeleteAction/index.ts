import AppListItemDeleteAction from './AppListItemDeleteAction'

export default AppListItemDeleteAction
