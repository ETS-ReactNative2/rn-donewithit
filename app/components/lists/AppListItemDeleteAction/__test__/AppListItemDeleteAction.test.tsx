import React from 'react'
import renderer from 'react-test-renderer'
import AppListItemDeleteAction from '../index'

jest.mock(
  '@expo/vector-icons/build/vendor/react-native-vector-icons/lib/create-icon-set.js',
  () => {
    return () => ''
  }
)

describe('AppListItemDeleteAction', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AppListItemDeleteAction />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
