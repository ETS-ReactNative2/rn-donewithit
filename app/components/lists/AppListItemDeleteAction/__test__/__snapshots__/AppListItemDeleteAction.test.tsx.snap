// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppListItemDeleteAction renders correctly 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  exclusive={true}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
>
  <View
    accessible={true}
    style={Object {}}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ff5252",
          "height": "100%",
          "justifyContent": "center",
          "width": 80,
        }
      }
    >
      <Text />
    </View>
  </View>
</RNGestureHandlerButton>
`;
